snuba/query/processors/prewhere.py

```diff
@@ -55,9 +64,60 @@ def process_query(self, query: Query, request_settings: RequestSettings,) -> Non
         )
         if prewhere_candidates:
             prewhere_conditions = [cond for _, cond in prewhere_candidates][
-                : self.__max_prewhere_conditions
+                :max_prewhere_conditions
             ]
             query.set_conditions(
                 list(filter(lambda cond: cond not in prewhere_conditions, conditions))
             )
         query.set_prewhere(prewhere_conditions)
+
+
+class CustomPrewhereProcessor(PrewhereProcessor):
+    def __init__(
+        self,
+        custom_candidates: Sequence[str],
```

Comment:
It'd probably be a bit easier to read/understand this parameter if it included "columns" in the name somewhere.